app/code/core/Mage/Checkout/controllers/CartController.php

```diff
@@ -677,7 +677,7 @@ public function ajaxUpdateAction()
                 if (!$quoteItem) {
                     Mage::throwException($this->__('Quote item is not found.'));
                 }
-                if ($qty == 0) {
+                if (is_numeric($qty) && $qty == 0) {
```

Comment:
```suggestion
                if (!is_numeric($qty)) {
                    Mage::throwException($this->__('Quantity is not numeric.'));
                }

                $qty = (float)$qty;
                if ($qty === 0.0) {
```

Non-numeric input is only validated/blocked by js right now(?). With an invalid input the item would be removed.

Cast to float is to match `setQty()` param type.

Reply:
Suggestion fixes that phpstan error too? (other errors can be fixed in another pr o/c)